### PR TITLE
Very basic authentication

### PR DESCRIPTION
**Pull request by hassan-bazzi**
Login returns a jwt token which we'll use in React 

Docs https://github.com/deepditch/deep.web/wiki/API-Docs-%F0%9F%9B%A0

### CODE REVIEW

File: app/Http/Controllers/AuthController.php [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    {
+        $request->validate([
+            'email' => 'required|email|unique:users,email|max:255',
+            'name' => 'required|max:255'
+        ]);
+
+        $user = User::create ([
+            'name' => $request->input('name'),
+            'email' => $request->input('email'),
+            'password' => Hash::make($request->input('password'))
```
Comment (drakesvoboda) [inline app/Http/Controllers/AuthController.php]: Is this salted?

File: app/Exceptions/Handler.php [hunk 2]
```diff
@@ -38,14 +43,28 @@ public function report(Exception $exception)
...
+            // detect previous instance
+            if ($exception->getPrevious() instanceof TokenExpiredException) {
+                return response()->json(['error' => 'TOKEN_EXPIRED'], $exception->getStatusCode());
+            } elseif ($exception->getPrevious() instanceof TokenInvalidException) {
+                return response()->json(['error' => 'TOKEN_INVALID'], $exception->getStatusCode());
+            } elseif ($exception->getPrevious() instanceof TokenBlacklistedException) {
+                return response()->json(['error' => 'TOKEN_BLACKLISTED'], $exception->getStatusCode());
+            } else {
+                return response()->json(['error' => "UNAUTHORIZED_REQUEST"], 401);
+            }
```
Comment (drakesvoboda) [inline app/Exceptions/Handler.php]: Can this become a switch statement

Comment (drakesvoboda) [reply app/Exceptions/Handler.php]: NVM, the instanceof buisness makes this more readable than a switch

File: app/Http/Controllers/AuthController.php (hunk 1 above)
Comment (hassan-bazzi) [reply app/Http/Controllers/AuthController.php]: Good question. Looks like it is: https://mnshankar.wordpress.com/2014/03/29/laravel-hash-make-explained/

**Pull request merged**